### ISSUE TITLE
add option 'with_recording' to waveform extractor select_units 

### ISSUE THREAD

**Issue by LiesDec** [SpikeInterface/spikeinterface#2281]
It would be useful to have the option 'with_recording' also in the method 'select_units' for the waveformextractor. 

On top of adding the parameter to [L766](https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/core/waveform_extractor.py#L766) and [L827](https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/core/waveform_extractor.py#L827) you would also need to copy the recording_info folder. 
For example modify lines [L801-L802 ](https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/core/waveform_extractor.py#L801-L802) to:

```
if self.has_recording():
    self.recording.dump(new_folder / "recording.json", relative_to=relative_to)
else:
    shutil.copytree(self.folder / "recording_info", new_folder / "recording_info")
```

There might be an extra check needed here as I understand that the 'recording_info' is not created in all spike interface versions.

**Comment by LiesDec** [comment]
@alejoe91  I see you labeled the issue as enhancement, but actually in my case it's more like a bug.

If I use the `select_units` method on a `we` object that has no recording attached (but has the 'recording_info' folder) it fails as it will try to load the recording which is not there. 

```
we_clean = we.select_units(keep_unit_ids, new_folder=datapath / 'raw' / animal / session_dt/ sorting_folder_name /'waveforms_clean')
```

```
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
Input In [7], in <module>
----> 1 we_clean = we.select_units(keep_unit_ids, new_folder=datapath / 'raw' / animal / session_dt/ sorting_folder_name /'waveforms_clean')

File ~/spikeinterface/src/spikeinterface/core/waveform_extractor.py:834, in WaveformExtractor.select_units(self, unit_ids, new_folder, use_relative_path)
    831             json.dump(check_json(new_sparsity.to_dict()), f)
    833     # we = WaveformExtractor.load(new_folder, with_recording = False)
--> 834     we = WaveformExtractor.load(new_folder)
    835 elif self.format == "zarr":
    836     raise NotImplementedError(
    837         "For zarr format, `select_units()` to a folder is not supported yet. "
    838         "You can select units in two steps:\n"
    839         "1. `we_new = select_units(unit_ids, new_folder=None)`\n"
    840         "2. `we_new.save(folder='new_folder', format='zarr')`"
    841     )

File ~/spikeinterface/src/spikeinterface/core/waveform_extractor.py:137, in WaveformExtractor.load(cls, folder, with_recording, sorting)
    135     return WaveformExtractor.load_from_zarr(folder, with_recording=with_recording, sorting=sorting)
    136 else:
--> 137     return WaveformExtractor.load_from_folder(folder, with_recording=with_recording, sorting=sorting)

File ~/spikeinterface/src/spikeinterface/core/waveform_extractor.py:177, in WaveformExtractor.load_from_folder(cls, folder, with_recording, sorting)
    175             pass
    176     if recording is None:
--> 177         raise Exception("The recording could not be loaded. You can use the `with_recording=False` argument")
    178     rec_attributes = None
    180 if sorting is None:

Exception: The recording could not be loaded. You can use the `with_recording=False` argument
```

If I then add the 'with_recording = False' option to line [L827](https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/core/waveform_extractor.py#L827) I get the error:

---------------------------------------------------------------------------
```
ValueError                                Traceback (most recent call last)
Input In [7], in <module>
----> 1 we_clean = we.select_units(keep_unit_ids, new_folder=datapath / 'raw' / animal / session_dt/ sorting_folder_name /'waveforms_clean')

File ~/spikeinterface/src/spikeinterface/core/waveform_extractor.py:833, in WaveformExtractor.select_units(self, unit_ids, new_folder, use_relative_path)
    830         with (new_folder / "sparsity.json").open("w") as f:
    831             json.dump(check_json(new_sparsity.to_dict()), f)
--> 833     we = WaveformExtractor.load(new_folder, with_recording = False)
    834     # we = WaveformExtractor.load(new_folder)
    835 elif self.format == "zarr":

File ~/spikeinterface/src/spikeinterface/core/waveform_extractor.py:137, in WaveformExtractor.load(cls, folder, with_recording, sorting)
    135     return WaveformExtractor.load_from_zarr(folder, with_recording=with_recording, sorting=sorting)
    136 else:
--> 137     return WaveformExtractor.load_from_folder(folder, with_recording=with_recording, sorting=sorting)

File ~/spikeinterface/src/spikeinterface/core/waveform_extractor.py:152, in WaveformExtractor.load_from_folder(cls, folder, with_recording, sorting)
    150 rec_attributes_file = folder / "recording_info" / "recording_attributes.json"
    151 if not rec_attributes_file.exists():
--> 152     raise ValueError(
    153         "This WaveformExtractor folder was created with an older version of spikeinterface"
    154         "\nYou cannot use the mode with_recording=False"
    155     )
    156 with open(rec_attributes_file, "r") as f:
    157     rec_attributes = json.load(f)

ValueError: This WaveformExtractor folder was created with an older version of spikeinterface
You cannot use the mode with_recording=False
```
Hence the fix I proposed above :)

**Comment by zm711** [comment]
Fixed by #2368.
